### PR TITLE
Updated Documents

### PR DESCRIPTION
**Pull request by JoseAlba**
## Thanks for starting a pull request on Material Components!
Added Hero images
Removed empty unneeded lines
Moved Types Header to new location
Changed HTML tags to MD
Removed commented out code
Kept consistency. Instead of switching between * and -
Keep headers consistent
There is no two headers in the same file with the same text. For better accessibility
Added Table of Contents page
Added Type content for documents that have multiple components

#### Don't forget:
- [x] Are you sure you mean this repo? This repository is a reference to the [main Flutter repository](https://github.com/flutter/flutter). Code-level issues should be created there.
- [ ] Link to GitHub issues it solves. ```closes #1234```
- [x] Sign the CLA bot. You can do this once the pull request is opened.

Be sure to read the [Flutter guide on contributing](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md).

### Related issues
 - Type `#` and the issue number or `#` and select the from the inline drop down.

### CODE REVIEW

File: docs/components/bottom_navigation.md [hunk 1]
```diff
@@ -7,18 +7,30 @@ iconId:
...
+  - [Contents](#contents)
+  - [Using bottom navigation](#using-bottom-navigation)
+  - [Bottom navigation example](#bottom-navigation-example)
+    - [Anatomy and key properties](#anatomy-and-key-properties)
+    - [Container attributes](#container-attributes)
+    - [Inactive icon attributes](#inactive-icon-attributes)
+    - [Inactive text label attributes](#inactive-text-label-attributes)
+    - [Active icon attributes](#active-icon-attributes)
+    - [Active text label attributes](#active-text-label-attributes)
+  - [Theming a bottom nav bar](#theming-a-bottom-nav-bar)
```
Comment (mingjane-work) [inline docs/components/bottom_navigation.md:28]: This should all be one level.

File: docs/components/bottom_navigation.md [hunk 2]
```diff
@@ -7,18 +7,30 @@ iconId:
...
+- [Bottom navigation](#bottom-navigation)
+  - [Contents](#contents)
+  - [Using bottom navigation](#using-bottom-navigation)
+  - [Bottom navigation example](#bottom-navigation-example)
+    - [Anatomy and key properties](#anatomy-and-key-properties)
+    - [Container attributes](#container-attributes)
+    - [Inactive icon attributes](#inactive-icon-attributes)
+    - [Inactive text label attributes](#inactive-text-label-attributes)
+    - [Active icon attributes](#active-icon-attributes)
+    - [Active text label attributes](#active-text-label-attributes)
```
Comment (mingjane-work) [inline docs/components/bottom_navigation.md:27]: This is more detail than we wanted for the TOC: please remove

File: docs/components/top_app_bars.md [hunk 3]
```diff
@@ -11,21 +11,26 @@ path: /catalog/material-top-app-bar/
...
+## Contents
 
-![Regular app bar: purple background, white text and icons and Contextual action bar: black background, white text and icons](assets/top_app_bars/Top-app-bars_types_side-by-side.png)
+- [Contents](#contents)
+- [Using the top app bar](#using-the-top-app-bar)
+- [Regular top app bar](#regular-top-app-bar)
+- [Contextual action bar](#contextual-action-bar)
+- [Theming a top app bar](#theming-a-top-app-bar)
+
+## Theming a top app bar](#theming-text-fields)
```
Comment (mingjane-work) [inline docs/components/top_app_bars.md:24]: Typo: missing a "["

File: docs/components/bottom_navigation.md (hunk 1 above)
Comment (JoseAlba) [reply docs/components/bottom_navigation.md:28]: Done

File: docs/components/bottom_navigation.md (hunk 2 above)
Comment (JoseAlba) [reply docs/components/bottom_navigation.md:27]: Done

File: docs/components/top_app_bars.md (hunk 3 above)
Comment (JoseAlba) [reply docs/components/top_app_bars.md:24]: Done

File: docs/components/bottom_app_bars.md [hunk 4]
```diff
@@ -13,28 +13,42 @@ path: /catalog/material-bottom-app-bar/
...
+## Contents
+
+- [Bottom app bars](#bottom-app-bars)
+  - [Contents](#contents)
+  - [Using the bottom app bar](#using-the-bottom-app-bar)
+  - [Bottom app bar example](#bottom-app-bar-example)
+  - [Anatomy and key properties](#anatomy-and-key-properties)
+    - [Container attributes](#container-attributes)
+    - [FAB attributes](#fab-attributes)
+    - [Icons attributes](#icons-attributes)
```
Comment (mingjane-work) [inline docs/components/bottom_app_bars.md]: We're not including this level of detail in the TOC

File: docs/components/bottom_app_bars.md [hunk 5]
```diff
@@ -13,28 +13,42 @@ path: /catalog/material-bottom-app-bar/
...
+  - [Theming](#theming)
+    - [Theming example](#theming-example)
```
Comment (mingjane-work) [inline docs/components/bottom_app_bars.md]: We're not including this level of detail in the TOC

File: docs/components/bottom_navigation.md (hunk 2 above)
Comment (mingjane-work) [inline docs/components/bottom_navigation.md]: We're not including this level of information in the TOC

File: docs/components/lists.md [hunk 6]
```diff
@@ -255,40 +262,44 @@ ListView(
...
 
+1. Supporting visuals
+1. Primary text
+1. Secondary text
+1. Tertiary text
+1. Metadata
+1. Controls
 
-##### Supporting visuals
+#### Supporting visuals for Three Line List
```
Comment (mingjane-work) [inline docs/components/lists.md:272]: FYI, Google titles use sentence case:

"This is an example of a Google title"
"This is Not an Example of a Google Title"

Comment (JoseAlba) [reply docs/components/lists.md:272]: Okay

**Comment by mingjane-work** [comment]
LGTM

File: docs/components/bottom_app_bars.md (hunk 4 above)
Comment (JoseAlba) [reply docs/components/bottom_app_bars.md]: Done

File: docs/components/bottom_app_bars.md (hunk 5 above)
Comment (JoseAlba) [reply docs/components/bottom_app_bars.md]: Done

File: docs/components/bottom_navigation.md (hunk 2 above)
Comment (JoseAlba) [reply docs/components/bottom_navigation.md]: Done

**Pull request merged**